napari/_qt/_qapp_model/_tests/test_qapp_model_menus.py

```diff
@@ -18,8 +18,14 @@ def test_build_qmodel_menu(qtbot, menu_id):
     with app.injection_store.register(
         providers={viewer.Viewer: lambda: mock, Window: lambda: mock}
     ):
+        init_qactions.cache_clear()
         init_qactions()
+
         menu = build_qmodel_menu(menu_id)
         qtbot.addWidget(menu)
-        # `>=` because separator bars count as actions
-        assert len(menu.actions()) >= len(app.menus.get_menu(menu_id))
+
+        # if there is no actions registered for this menu, this is likely a placeholder submenu
+        # for example tools/acquisition, these placeholder submenus do not show up under app.menus
+        if len(menu.actions()) != 0:
```

Comment:
I would much prefer to see this test skipped explicitly for the tools menu which afaik is the only one that may reasonably have 0 actions. I'm also not sure what you mean by placeholder submenus? Why do they not show up under `app.menus`?

Reply:
under app model, sub menus and menus are treated differently, they both have an ID that is declared under menu constants but they end up being created as different classes, which is why app_model.menus() do not show these, as menus are MenuItem stored under _menu_items in the app model, and submenus exists within MenuItem as SubMenuItem

I am fine changing the check to exclude tools menu